### PR TITLE
Reduce connection from heartbeet and rsyslog configurer

### PR DESCRIPTION
**Pull request by TheRealHaoLiu**
##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 22.0.1.dev51+ge91725819b
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by TheRealHaoLiu** [comment]
verify that heart beat still get sent every 15 second with 
```python
with pg_bus_conn(new_connection=True) as conn:
    conn.listen("web_heartbeet")
    for e in conn.events(yield_timeouts=True):
        if e is not None:
            received_timestamp = time.time()
            body = json.loads(e.payload)
            print(received_timestamp, body)
```
output
```
1681310508.6908503 {'hostname': 'awx_1', 'ip': None, 'action': 'online'}
1681310523.7092388 {'hostname': 'awx_1', 'ip': None, 'action': 'online'}
1681310538.8152282 {'hostname': 'awx_1', 'ip': None, 'action': 'online'}
1681310553.8322735 {'hostname': 'awx_1', 'ip': None, 'action': 'online'}
1681310568.8498118 {'hostname': 'awx_1', 'ip': None, 'action': 'online'}
1681310583.867774 {'hostname': 'awx_1', 'ip': None, 'action': 'online'}
1681310598.9887488 {'hostname': 'awx_1', 'ip': None, 'action': 'online'}
```

**Comment by TheRealHaoLiu** [comment]
verify that rsyslog configurer still function using 
```
tools_awx_1       | 2023-04-12 14:46:30,074 INFO     [-] awx.main.rsyslog_configurer Change in logging settings found. Restarting rsyslogd
```

and verified that rsyslog configuration was updated 
```
bash-5.1$ cat /var/lib/awx/rsyslog/rsyslog.conf
$WorkDirectory /var/lib/awx/rsyslog
$MaxMessageSize 700000
$IncludeConfig /var/lib/awx/rsyslog/conf.d/*.conf
main_queue(queue.spoolDirectory="/var/lib/awx" queue.maxdiskspace="1g" queue.type="Disk" queue.filename="awx-external-logger-backlog")
module(load="imuxsock" SysSock.Use="off")
input(type="imuxsock" Socket="/var/run/awx-rsyslog/rsyslog.sock" unlink="on" RateLimit.Burst="0")
template(name="awx" type="string" string="%rawmsg-after-pri%")
action(type="omfile" file="/dev/null")
bash-5.1$ cat /var/lib/awx/rsyslog/rsyslog.conf
$WorkDirectory /var/lib/awx/rsyslog
$MaxMessageSize 700000
$IncludeConfig /var/lib/awx/rsyslog/conf.d/*.conf
main_queue(queue.spoolDirectory="/var/lib/awx" queue.maxdiskspace="1g" queue.type="Disk" queue.filename="awx-external-logger-backlog")
module(load="imuxsock" SysSock.Use="off")
input(type="imuxsock" Socket="/var/run/awx-rsyslog/rsyslog.sock" unlink="on" RateLimit.Burst="0")
template(name="awx" type="string" string="%rawmsg-after-pri%")
module(load="omhttp")
action(type="omhttp" server="localhost" serverport="400" usehttps="off" allowunsignedcerts="off" skipverifyhost="off" action.resumeRetryCount="-1" template="awx" action.resumeInterval="5" errorfile="/var/log/tower/rsyslog.err")
```

**Comment by TheRealHaoLiu** [comment]
```
awx=# SELECT application_name FROM pg_stat_activity;
-[ RECORD 1 ]----+----------------------------------------------------------------
application_name |
-[ RECORD 2 ]----+----------------------------------------------------------------
application_name |
-[ RECORD 3 ]----+----------------------------------------------------------------
application_name | psql
-[ RECORD 4 ]----+----------------------------------------------------------------
application_name | awx_1-319-manage.py run_cache_clear
-[ RECORD 5 ]----+----------------------------------------------------------------
application_name | awx_1-2409-/var/lib/awx/venv/awx/bin/daphne -b 127.0.0.1 -p 805
-[ RECORD 6 ]----+----------------------------------------------------------------
application_name | awx_1-2406-manage.py run_heartbeet
-[ RECORD 7 ]----+----------------------------------------------------------------
application_name | awx_1-2401-manage.py run_dispatcher
-[ RECORD 8 ]----+----------------------------------------------------------------
application_name | awx_1-2401-manage.py run_dispatcher
-[ RECORD 9 ]----+----------------------------------------------------------------
application_name | awx_1-2407-manage.py run_rsyslog_configurer
-[ RECORD 10 ]---+----------------------------------------------------------------
application_name | awx_1-2406-manage.py run_heartbeet
-[ RECORD 11 ]---+----------------------------------------------------------------
application_name | awx_1-2408-manage.py run_wsrelay
-[ RECORD 12 ]---+----------------------------------------------------------------
application_name | awx_1-2407-manage.py run_rsyslog_configurer
-[ RECORD 13 ]---+----------------------------------------------------------------
application_name | awx_1-2401-manage.py run_dispatcher
-[ RECORD 14 ]---+----------------------------------------------------------------
application_name | awx_1-2410-manage.py run_callback_receiver
-[ RECORD 15 ]---+----------------------------------------------------------------
application_name |
-[ RECORD 16 ]---+----------------------------------------------------------------
application_name |
-[ RECORD 17 ]---+----------------------------------------------------------------
application_name |

```

rsyslog_configurer and heartbeet no longer hold 2 connections

File: awx/main/management/commands/run_rsyslog_configurer.py [hunk 1]
```diff
@@ -26,6 +27,7 @@ def handle(self, *arg, **options):
                 conn.listen("rsyslog_configurer")
                 # reconfigure rsyslog on start up
                 reconfigure_rsyslog()
+                connection.close()
                 for e in conn.events(yield_timeouts=True):
                     if e is not None:
                         logger.info("Change in logging settings found. Restarting rsyslogd")
```
Comment (AlanCoding) [inline awx/main/management/commands/run_rsyslog_configurer.py:33]: I still think it would be good to close again after `TaskWorker.run_callable(body)`

**Comment by AlanCoding** [comment]
I think we should look at `awx-manage run_cache_clear` too. The only reason we haven't looked at it is because it doesn't run in the docker-compose environment, but it will still be relevant in normal OCP.

Similarly to the rsyslog configurer, it likely is connected (aside from listener connection) at startup and uses the database to run its task. So it would be best to close the connection before entering the main loop and every time after the task finishes.

Because it only runs tasks infrequently, there is no coherent argument for keeping the connection open for better high-traffic performance.

**Pull request closed without merging**